upstream.go

```diff
@@ -206,9 +206,28 @@ func (uc *upstreamConn) handleMessage(msg *irc.Message) error {
 	case "NOTICE":
 		uc.logger.Print(msg)
 
-		uc.forEachDownstream(func(dc *downstreamConn) {
-			dc.SendMessage(msg)
-		})
+		if msg.Prefix.User == "" && msg.Prefix.Host == "" {
+			uc.SendMessage(msg)
+		} else {
+			var nick, text string
+			if err := parseMessageParams(msg, &nick, &text); err != nil {
+				return err
+			}
+
+			target := nick
+			if nick == uc.nick {
+				target = msg.Prefix.Name
+			}
+			uc.Logfn(target, "<%s> %s", msg.Prefix.Name, text)
```

Comment:
This belongs to a later commit.

Reply:
Ah, it seems like this has been removed in 3d8819ad8054f25aa83a83a23208f5a70c0b619f.